packages/instant/src/containers/latest_error.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+import { connect } from 'react-redux';
+
+import { SlidingError } from '../components/sliding_error';
+import { State } from '../redux/reducer';
+import { errorDescription } from '../util/error_description';
+
+export interface LatestErrorComponentProps {
+    assetData?: string;
+    latestError?: any;
+    latestErrorDismissed?: boolean;
+}
+
+export const LatestErrorComponent: React.StatelessComponent<LatestErrorComponentProps> = props => {
+    if (!props.latestError) {
+        return <div />;
+    }
+    const slidingDirection = props.latestErrorDismissed ? 'down' : 'up';
+    const { icon, message } = errorDescription(props.latestError, props.assetData);
+    return <SlidingError direction={slidingDirection} icon={icon} message={message} />;
+};
```

Comment:
A convention I'm trying to follow is to not have any Component code in these container files, they simply connect a pure presentational component to the redux store and do any state -> prop mapping. Seems like you don't need this component to me?

- The sliding direction logjc can just be done in mapStateToProps.
- icon is a pure function of state.
- message is a pure function of state.

The only thing is not rendering when `latestError` is not defined, but do we need that? The error hides behind container.

Reply:
You could change it so that `latestError` sticks around and you simply toggle it up and down in the flasher / redux. This kind of seems better to me because then it becomes impossible to make the error just disappear?